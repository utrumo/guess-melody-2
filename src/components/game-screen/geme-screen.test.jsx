import React from 'react';
import renderer from 'react-test-renderer';
import createAudioNodeMock from '../../mocks/audio-node-mock';
import GameScreen from './game-screen.jsx';

const questions = [
  {
    type: `artist`,
    song: {
      artist: `Quincas Moreira`,
      src: `https://es31-server.appspot.com/guess-melody/static/music/Blue_Whale.mp3`
    },
    answers: [
      {
        picture: `https://es31-server.appspot.com/guess-melody/static/artist/Quincas_Moreira.jpg`,
        artist: `Quincas Moreira`
      },
      {
        picture: `https://es31-server.appspot.com/guess-melody/static/artist/Density_n_Time.jpg`,
        artist: `Density & Time`
      },
      {
        picture: `https://es31-server.appspot.com/guess-melody/static/artist/Endless_Love.jpg`,
        artist: `Endless Love`
      }
    ]
  },
  {
    type: `genre`,
    genre: `reggae`,
    answers: [
      {
        src: `https://es31-server.appspot.com/guess-melody/static/music/Addis_Ababa.mp3`,
        genre: `reggae`
      },
      {
        src: `https://es31-server.appspot.com/guess-melody/static/music/Azure.mp3`,
        genre: `electronic`
      },
      {
        src: `https://es31-server.appspot.com/guess-melody/static/music/Whaling_City.mp3`,
        genre: `country`
      },
      {
        src: `https://es31-server.appspot.com/guess-melody/static/music/Skanada.mp3`,
        genre: `reggae`
      }
    ]
  }
];
const maxMistakes = 3;
const mistakes = 2;
const step = 0;
const time = 30000;
const artistQuestion = questions[0];
const genreQuestion = questions[1];

it(`GameScreen renders artist question`, () => {
  const tree = renderer
    .create((
      <GameScreen
        maxMistakes={maxMistakes}
        mistakes={mistakes}
        time={time}
        step={step}
        question={artistQuestion}
        onAnswer={jest.fn()}
        onUserRestart={jest.fn()}
      />
    ),
    createAudioNodeMock
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`GameScreen renders genre question`, () => {
  const tree = renderer
    .create((
      <GameScreen
        maxMistakes={maxMistakes}
        mistakes={mistakes}
        time={time}
        step={step}
        question={genreQuestion}
        onAnswer={jest.fn()}
        onUserRestart={jest.fn()}
      />
    ),
    createAudioNodeMock
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
