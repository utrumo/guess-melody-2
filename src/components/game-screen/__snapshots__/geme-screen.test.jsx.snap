// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameScreen renders artist question 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      onClick={[MockFunction]}
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx="390"
        cy="390"
        r="370"
        style={
          Object {
            "filter": "url(#blur)",
            "transform": "rotate(-90deg) scaleY(-1)",
            "transformOrigin": "center",
          }
        }
      />
    </svg>
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        00
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        30
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="correct"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
      </div>
    </div>
    <form
      className="game__artist"
    >
      <div
        className="artist"
      >
        <button
          className="artist__input visually-hidden"
          id="answer-0"
          name="answer"
          onClick={[Function]}
          type="button"
        />
        <label
          className="artist__name"
          htmlFor="answer-0"
        >
          <img
            alt="Quincas Moreira"
            className="artist__picture"
            src="https://es31-server.appspot.com/guess-melody/static/artist/Quincas_Moreira.jpg"
          />
          Quincas Moreira
        </label>
      </div>
      <div
        className="artist"
      >
        <button
          className="artist__input visually-hidden"
          id="answer-1"
          name="answer"
          onClick={[Function]}
          type="button"
        />
        <label
          className="artist__name"
          htmlFor="answer-1"
        >
          <img
            alt="Density & Time"
            className="artist__picture"
            src="https://es31-server.appspot.com/guess-melody/static/artist/Density_n_Time.jpg"
          />
          Density & Time
        </label>
      </div>
      <div
        className="artist"
      >
        <button
          className="artist__input visually-hidden"
          id="answer-2"
          name="answer"
          onClick={[Function]}
          type="button"
        />
        <label
          className="artist__name"
          htmlFor="answer-2"
        >
          <img
            alt="Endless Love"
            className="artist__picture"
            src="https://es31-server.appspot.com/guess-melody/static/artist/Endless_Love.jpg"
          />
          Endless Love
        </label>
      </div>
    </form>
  </section>
</section>
`;

exports[`GameScreen renders genre question 1`] = `
<section
  className="game game--genre"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      onClick={[MockFunction]}
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx="390"
        cy="390"
        r="370"
        style={
          Object {
            "filter": "url(#blur)",
            "transform": "rotate(-90deg) scaleY(-1)",
            "transformOrigin": "center",
          }
        }
      />
    </svg>
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        00
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        30
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="correct"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Выберите 
      reggae
       треки
    </h2>
    <form
      className="game__tracks"
      onSubmit={[Function]}
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            checked={false}
            className="game__input visually-hidden"
            id="answer-0"
            name="answer"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="game__check"
            htmlFor="answer-0"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            checked={false}
            className="game__input visually-hidden"
            id="answer-1"
            name="answer"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="game__check"
            htmlFor="answer-1"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            checked={false}
            className="game__input visually-hidden"
            id="answer-2"
            name="answer"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="game__check"
            htmlFor="answer-2"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            checked={false}
            className="game__input visually-hidden"
            id="answer-3"
            name="answer"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="game__check"
            htmlFor="answer-3"
          >
            Отметить
          </label>
        </div>
      </div>
      ;
      <button
        className="game__submit button"
        type="submit"
      >
        Ответить
      </button>
    </form>
  </section>
</section>
`;
